[DOCU-1549] Updates immunity download doc with new version compatibility information (#2932)
Kong/docs.konghq.com: app/enterprise/1.3-x/brain-immunity/install-configure.md

* **Collector App** enables communication between Kong Enterprise and Brain and Immunity. The Kong Collector App comes with your purchase of Bran and Immunity.
* **Collector Plugin** enables communication between Kong Enterprise and Kong Collector App. The Kong Collector Plugin comes with your purchase of Brain and Immunity.

## Version Compatibility

> Kong Brain and Kong Immunity are not compatible with Kong Enterprise v1.3.x. To use Kong Brain and Immunity, you need to upgrade to Kong Enterprise v1.5.x or above.

Reviewer: There's a bunch of docs for Brain and Immunity though, which shows that at one point, 1.3-x was compatible. I wonder if this section also needs to say that Brain and Immunity is deprecated and not available for download anymore. Something like:

```suggestion
Kong Brain is deprecated, and Kong Immunity is no longer compatible with Kong Enterprise v1.3-x. To use Kong Immunity, you need to upgrade to Kong Enterprise v1.5.x or above.
```
